src/lib/describe-endpoints.ts

```diff
@@ -0,0 +1,239 @@
+import { Endpoint } from "../utils/types";
+import { methodDetailsToString, schemaToString, truncateExample, capitalizeFirstLetter, getParameterExample, getResponseParameterExample } from "./description-helpers/endpoint-parsers" //getQueryParameterExample 
+import { endpointSystemPrompt, parameterSystemPrompt, endpointDescriptionPrompt, parameterDescriptionPrompt } from "./description-helpers/prompts";
+import callOpenAI from "./callOpenAI";
+
+
+export async function describeApiEndpoint(endpoint: Endpoint, model: string): Promise<string | null> {
+
+  const methodsString = Object.entries(endpoint.data.methods)
+    .map(([method, details]) => `${method.toUpperCase()}: ` + methodDetailsToString(details))
+    .join('\n');
+
+  let endpointPrompt = endpointDescriptionPrompt(methodsString);
+
+  try {
+    const endpointDescription = await callOpenAI(endpointPrompt, endpointSystemPrompt, model);
+    if (endpointDescription.choices && endpointDescription.choices.length > 0 && endpointDescription.choices[0].message) {
+      
+      const content = endpointDescription.choices[0].message.content;
+      const capContent = capitalizeFirstLetter(content);
+      
+      return capContent;
+    } else {
+      console.warn('Unexpected OpenAI response format:', endpointDescription);
+      return null;
+    }
+  } catch (error) {
+    console.error('Error calling OpenAI:', error);
+    return null;
+  }
+}
+
+export async function describeRequestBodySchemaParameters(endpoint: Endpoint, endpointDescription: string, model: string): Promise<Record<string, string | null>> {
+  const parameterDescriptions: Record<string, string | null> = {};
+  const mostRecentExamples: Record<string, any> = {};
```

Comment:
Try `npm run lint` :grimacing: 

Options:
1. Fix the instances of `any` - most are not super complicated (just need to learn about how to `Pick`, `Extract`, `Omit` and index from existing types), some are a little tricky, though - you may need to use generics for some
2. Add `@typescript-eslint/no-explicit-any` to ignore list

I would lean towards 2 this since I don't imagine 1 will be a quick job for you.

Indexing from existing types:
```
type Example = {
 a: {
  b: string
} 
}
type l = Example["a"];
```